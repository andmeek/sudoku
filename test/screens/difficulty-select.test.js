import DifficultySelect from '../../src/screens/difficulty-select.vue'
import { shallowMount } from '@vue/test-utils'

describe('DifficultySelect.vue', () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallowMount(DifficultySelect)
  })

  test('matches snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })

  test('emits a difficultyselected event when the user picks one', () => {
    wrapper.findAll('button').at(0).trigger('click')

    expect(wrapper.emitted().difficultyselected).toBeTruthy()
    expect(wrapper.emitted().difficultyselected[0]).toContain('Easy')

    wrapper.findAll('button').at(2).trigger('click')

    expect(wrapper.emitted().difficultyselected[1]).toContain('Hard')
  })

  test('emits an exit event when the user clicks back', () => {
    wrapper.find({ ref: 'exit' }).trigger('click')

    expect(wrapper.emitted().exit).toBeTruthy()
  })
})
