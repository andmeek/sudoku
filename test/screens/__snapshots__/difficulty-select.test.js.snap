// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DifficultySelect.vue matches snapshot 1`] = `
<div class="difficulty screen">
  <h1>Select Difficulty</h1>
  <div class="actions"><button class="menu">
      Easy
    </button> <button class="menu">
      Medium
    </button> <button class="menu">
      Hard
    </button> <button class="menu">
      Very Hard
    </button> <button class="menu">
      Insane
    </button> <button class="menu">
      Unthinkable
    </button></div> <button class="menu">
    Back
  </button>
</div>
`;
